server/utils/statusTag.ts

```diff
@@ -0,0 +1,81 @@
+import { ApprovedPremisesApplicationStatus as ApplicationStatus } from '../@types/shared'
+
+export type StatusTagOptions = { addLeftMargin?: boolean; showOnOneLine?: boolean }
+type StatusColour = 'green' | 'grey' | 'blue' | 'yellow' | 'red' | 'pink' | ''
+type Status = ApplicationStatus
+export class StatusTag<T extends Status> {
```

Comment:
Could this be an `interface` instead of a class? I don't think it gets used on it's own

Reply:
Yeah I did think about this, it should be an abstract class. The issue is we then loose the ability to test the `html` method logic for all subclasses because you can't instantiate an abstract class to test it. 
I think a solution would be to extract the `html` logic to a seperate function that `html` simply calls... I'll give it a go